modules/network/main.tf

```diff
@@ -0,0 +1,177 @@
+  availability_zone = "us-east-1b"
+
+  tags = {
+    Name = "Private Subnet 2"
+  }
+}
+
+resource "aws_route_table" "public" {
+  vpc_id = aws_vpc.main.id
+
+  tags = {
+    Name = "${var.env_name}-public-routing-table"
+  }
+}
+
+resource "aws_route" "public" {
+  route_table_id         = aws_route_table.public.id
+  destination_cidr_block = "0.0.0.0/0"
+  gateway_id             = aws_internet_gateway.main.id
+}
+
+resource "aws_main_route_table_association" "main" {
+  vpc_id         = aws_vpc.main.id
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "public_1" {
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public_1.id
+}
+
+resource "aws_route_table_association" "public_2" {
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public_2.id
+}
+
+resource "aws_security_group" "alb_main" {
+  name        = "alb_security_group"
+  description = "Security group for ALB"
+  vpc_id      = aws_vpc.main.id
+
+  tags = {
+    Name = "alb_security_group"
+  }
+}
+
+resource "aws_security_group_rule" "allow_incoming_http" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 80
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+  security_group_id = aws_security_group.alb_main.id
+}
+
+resource "aws_security_group_rule" "allow_incoming_https" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 443
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+  security_group_id = aws_security_group.alb_main.id
+}
+
+resource "aws_security_group_rule" "outgoing_to_anywhere" {
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.alb_main.id
+}
+
+resource "aws_lb" "alb_ecs_main" {
+  name               = "alb-ecs-main"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.alb_main.id]
+  subnets            = [aws_subnet.public_1.id, aws_subnet.public_2.id]
+}
+
+resource "aws_security_group" "ecs_main" {
+  name        = "ecs_security_group"
+  description = "Security group for ECS"
+  vpc_id      = aws_vpc.main.id
+
+  ingress {
+    from_port       = 0
+    to_port         = 0
+    protocol        = "-1"
+    security_groups = [aws_security_group.alb_main.id]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "${var.env_name}-ecs-sg"
+  }
+}
+
+resource "aws_alb_target_group" "main" {
+  name        = "alb-ecs-target-group"
+  port        = 80
+  protocol    = "HTTP"
+  target_type = "ip"
+  vpc_id      = aws_vpc.main.id
+}
+
+resource "aws_lb_listener" "app_http" {
```

Comment:
this is not really part of the network module IMO, it could be belongs to the `ALB module` instead

Reply:
edit: I will refactor this when I refactor everything else if this is OK?